plugins/base/src/main/kotlin/renderers/html/NavigationDataProvider.kt

```diff
@@ -0,0 +1,92 @@
+package org.jetbrains.dokka.base.renderers.html
+
+import org.jetbrains.dokka.analysis.PsiDocumentableSource
+import org.jetbrains.dokka.base.renderers.sourceSets
+import org.jetbrains.dokka.base.transformers.documentables.isException
+import org.jetbrains.dokka.model.*
+import org.jetbrains.dokka.pages.*
+import org.jetbrains.dokka.plugability.DokkaContext
+import org.jetbrains.dokka.transformers.pages.PageTransformer
+
+open class NavigationPageInstaller(val context: DokkaContext) : NavigationDataProvider(), PageTransformer {
```

Comment:
The class was getting a bit too long for `htmlPreprocessors.kt`, extracted it. Will point out which parts are new

Reply:
I don't think it necessarily _should_ be in the same file, sooner or later it'll grow so big that all of them will have to be extracted into their own ones. Imho, grouping classes by features rather than domain is easier for navigating around

Doesn't make a huge difference now, so returned it

Reply:
Mu concern: If I open `htmlPreprocessors.kt` I expect to see ALL preprocessors there.

> Imho, grouping classes by features rather than domain is easier for navigating around

Anyway now there are two different grouping. It is misleading.